crates/bevy_ecs/src/entity/hash_set.rs

```diff
@@ -0,0 +1,364 @@
+use core::{
+    fmt::{self, Debug, Formatter},
+    iter::FusedIterator,
+    marker::PhantomData,
+    ops::{
+        BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Deref, DerefMut, Sub,
+        SubAssign,
+    },
+};
+
+#[cfg(feature = "bevy_reflect")]
+use bevy_reflect::Reflect;
+use bevy_utils::hashbrown::hash_set::{self, HashSet};
+
+use super::{Entity, EntityHash, EntitySetIterator};
+
+/// A [`HashSet`] pre-configured to use [`EntityHash`] hashing.
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect))]
+#[derive(Debug, Clone, Default)]
+pub struct EntityHashSet(pub(crate) HashSet<Entity, EntityHash>);
+
+impl EntityHashSet {
+    /// Creates an empty `EntityHashSet`.
+    ///
+    /// Equivalent to [`HashSet::with_hasher(EntityHash)`].
+    ///
+    /// [`HashSet::with_hasher(EntityHash)`]: HashSet::with_hasher
+    pub fn new() -> Self {
+        Self(HashSet::with_hasher(EntityHash))
+    }
+
+    /// Creates an empty `EntityHashSet` with the specified capacity.
+    ///
+    /// Equivalent to [`HashSet::with_capacity_and_hasher(n, EntityHash)`].
+    ///
+    /// [`HashSet::with_capacity_and_hasher(n, EntityHash)`]: HashSet::with_capacity_and_hasher
+    pub fn with_capacity(n: usize) -> Self {
+        Self(HashSet::with_capacity_and_hasher(n, EntityHash))
+    }
+
+    /// Returns the inner [`HashSet`].
+    pub fn into_inner(self) -> HashSet<Entity, EntityHash> {
+        self.0
+    }
+
+    /// Clears the set, returning all elements in an iterator.
+    ///
+    /// Equivalent to [`HashSet::drain`].
+    pub fn drain(&mut self) -> Drain<'_> {
```

Comment:
Do you also want a wrapper for `ExtractIf`?

Reply:
yes!